rules/prebuilt_frameworks_importer.bzl

```diff
@@ -0,0 +1,177 @@
+    Args:
+        root: The root path where the BUILD.bazel file will be created
+        frameworks_subpath: The path from the root where the frameworks are to be found
+    """
+
+    buildfile_content = []
+
+    binaries_path = root
+    for element in frameworks_subpath.split("/"):
+        binaries_path = binaries_path.get_child(element)
+
+    if binaries_path.exists:
+        frameworks = [path for path in binaries_path.readdir() if str(path).endswith(".framework")]
+        for framework in frameworks:
+            new_content = """\
+filegroup(
+    name = "%s", 
+    srcs = glob(["%s/%s/**/*"]),
+    visibility = ["//visibility:public"],
+)\
+            """ % (framework.basename.split(".framework")[0], frameworks_subpath, framework.basename)
+            buildfile_content.append(new_content)
+
+    return buildfile_content
+
+def _copy_files(ctx, files, destination):
+    """Copy the specified files to the destination"""
+
+    for file in files:
+        _execute(ctx, "cp -r \"%s\" \"%s\"" % (file, destination))
+
+def _execute(ctx, cmd):
+    """Execute the specified command"""
+
+    result = ctx.execute(["sh", "-c", cmd], quiet = ctx.attr.quiet)
+    if result.return_code != 0:
+        fail(
+            """
+====== Command '%s' failed ======
+====== stdout ======
+%s
+====== stderr ======
+%s
+        """ % (cmd, result.stdout, result.stderr),
+        )
+
+def _get_absolute_paths(ctx, file_labels):
+    """Return the absolut paths for the files"""
+
+    return [ctx.path(file_label) for file_label in file_labels]
+
+###############
+# Carthage
+###############
+
+def carthage(
+        name,
+        directory = "",
+        files = ["Cartfile"],
+        cmd = "carthage bootstrap --no-use-binaries --platform iOS",
+        quiet = quiet_default):
+    prebuilt_frameworks_importer(
+        implementation = _carthage_impl,
+        name = name,
+        directory = directory,
+        files = files,
+        cmd = cmd,
+        quiet = quiet,
+    )
+
+def _carthage_impl(ctx):
+    wd = ctx.path(".")
+
+    absolut_files = _get_absolute_paths(ctx, ctx.attr.file_labels)
+    _copy_files(ctx, absolut_files, wd)
+    _execute(ctx, ctx.attr.cmd)
+
+    buildfile_content = []
+
+    frameworks_subpath = "Carthage/Build"
+    for path in ctx.path(frameworks_subpath).readdir():  # Carthage organizes the built platforms in folders
+        if str(path).endswith(".version"):
+            continue  # Skip version files
+
+        dynamic_frameworks_subpath = frameworks_subpath + "/" + path.basename
+        buildfile_content = buildfile_content + _create_buildfile_content(wd, dynamic_frameworks_subpath)
+
+        static_frameworks_subpath = dynamic_frameworks_subpath + "/Static"
+        buildfile_content = buildfile_content + _create_buildfile_content(wd, static_frameworks_subpath)
+
+    ctx.file(
+        "BUILD.bazel",
+        "\n".join(buildfile_content),
+    )
+
+###############
+# Cocoapods
+###############
+
+def cocoapods(
```

Comment:
should this be a bit more verbose? `import_cocoapods_frameworks` or something like that?

Reply:
Sure. Instead of import I would go for `build_cocoapods_frameworks`, or `prebuild_cocoapods_frameworks`. What do you think?

Reply:
Gone with `build_cocoapods_frameworks`